app/src/main/java/com/example/myfirstapp/ShoppingTabFragment.java

```diff
@@ -2,16 +2,121 @@
 
 import android.os.Bundle;
 import android.support.v4.app.Fragment;
+import android.support.v4.content.ContextCompat;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.text.Editable;
+import android.text.InputType;
+import android.text.TextWatcher;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
 
 /**
  */
 public class ShoppingTabFragment extends Fragment {
+
+    // Adapter
+    SimpleAdapter recyclerAdapter;
+
+    // RecyclerView
+    RecyclerView recyclerView;
+
+    // Field for entering item
+    EditText editShoppingItemName;
+
+    // Field for entering price of item
+    EditText editShoppingItemPrice;
+
+    // Title of the list area
+    TextView listTitle;
+
+    // Button for adding item
+    Button addItem;
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle bundle) {
-        return inflater.inflate(R.layout.fragment_shopping_tab, container, false);
+        super.onCreate(bundle);
+        final View view = inflater.inflate(R.layout.fragment_shopping_tab, container, false);
+
+        editShoppingItemName = view.findViewById(R.id.editItemName);
```

Comment:
Do you want to keep things like this "pure" or are you open to library like [Butter Knife](http://jakewharton.github.io/butterknife/)?

Reply:
I don't know what that difference would be between keeping it "pure" or using Butter Knife. Though Butter Knife is on my list of things to learn so let's do that.

Reply:
word!